### ISSUE TITLE
Render vehicles always in high detail effects player visible range

### ISSUE THREAD

**Issue by gptrk0** [multitheftauto/mtasa-blue#708]
**Describe the bug**
The maximum draw distance is 300m.
BUT: If I enable 'Render vehicles always in high detail', I can see players ~1000m long.
I think it's a bug. (or not?)

**To reproduce**
1. Go to Settings, Video tab.
2. Enable 'Render vehicles always in high detail' and save it.
3. Restart MTA.
4. Connect to a server.

**Comment by ArranTuna** [comment]
And where is a screenshot of seeing a player streamed in 1000 m away?

**Comment by gptrk0** [comment]
![image](https://user-images.githubusercontent.com/923242/48731531-2bf99980-ec35-11e8-852d-f9c88f20d349.png)

**Comment by ArranTuna** [comment]
Those images show a player at the same distance, you're saying that you can see a player from 1000 meters away. Which I'm pretty sure would be impossible as at that distance they would be the size of less than 1 pixel. Oh and the collisions of what they're stood on wouldn't be applied to an object 1000 meters away so they'd fall through the ground.

**Comment by gptrk0** [comment]
Not the same.
They are on the two endpoints.

If the setting is enabled, I can see the peds far away.

(Maybe not 1000m, but minimum 2x bigger)

**Comment by Dutchman101** [comment]
> ![image](https://user-images.githubusercontent.com/923242/50401611-e70ed880-0787-11e9-87fc-6ae2b8684afb.png)


The distance between you and the test player clearly doesn't match between your 2 screenshots, therefore this test is invalid. Distance can clearly be distinguished by the amount of road markings passed, I wonder why manipulation like this is neccesary when you're confident that this issue exists. It's nothing to do with 'endpoints' because the purpose of your test is a comparisation between the HQ lod setting being on and off.

Furthermore, I cannot reproduce this issue. Please provide a solid reproduction/practical test

**Comment by gptrk0** [comment]
> Furthermore, I cannot reproduce this issue. Please provide a solid reproduction/practical test

Did you tried that...?
And you restarted the MTA, when changed the setting?

**Comment by Dutchman101** [comment]
> > Furthermore, I cannot reproduce this issue. Please provide a solid reproduction/practical test
> 
> Did you tried that...?
> And you restarted the MTA, when changed the setting?

Yes.. of course we follow our own reproduction guidelines.

**Comment by gptrk0** [comment]
I gonna make a video...

**Comment by StifflersMom** [comment]
I can reprouce it - The distance when a player become visible ist ~ twice as high with "Render vehicles always in high detail" - Exactly with the provided steps to reproduce.

**Comment by StifflersMom** [comment]
I made 2 videos of it. The ped on the runway is only for mark the higher streaming distance.
Target is the player which is standing on the roof of the building.

without Render vehicles always in high detail:
https://www.youtube.com/watch?v=kFQ2x4tEIjI

with Render vehicles always in high detail:
https://www.youtube.com/watch?v=TkhlSoaazSQ

**Comment by K13O** [comment]
I was able to reproduce it as well. Some people on my server started calling a guy hacker because he could see then from a longer distance. Now we know why.

**Comment by TheRadioGuy** [comment]
+1

I can reproduce it, too

**Comment by samr46** [comment]
It's caused by unfinished `high_detail_peds` option from PR #231.

CSettingsSA.cpp | CSettingsSA::ResetPedsLODDistance()
```
bool bHighDetailPeds;
g_pCore->GetCVars()->Get("high_detail_peds", bHighDetailPeds);
if (bHighDetailPeds)
{
    ms_fPedsLODDistance = MAX_PEDS_LOD_DISTANCE;
}
```

`if (bHighDetailPeds)` will be true if high_detail_peds is empty in coreconfig.xml (which is this case because default value isn't defined in CClientVariables.cpp). By the way to avoid such unexpected behavior simple `== true` is required.

Also i think that this option is a mistake (players shouldn't be able to set peds lod distance to maximum, it can give a real advantage in PVP if this distance isn't managed by server owner). Maybe it's better to remove it.

**Comment by Einheit-101** [comment]
I am using setPedsLODDistance(90) + vehicles always high detail, that is so far the best value for competitive gameplay, anything higher causes the players to stream in before most objects get streamed in, which, in combination with the bad MTA streamer, causes a mess because some players start setting render distance to low and shoot everyone through the whole map.

**Comment by Dutchman101** [comment]
This is a confirmed issue and (as far I'm aware) it hasn't been fixed yet, so please don't close it on your own @galeapatrik, thanks.

**Comment by qaisjp** [comment]
Fixed by #832

**Comment by gptrk0** [comment]
Thanks guys.
